mllib/src/main/scala/org/apache/spark/mllib/clustering/PowerIterationClustering.scala

```diff
@@ -397,11 +409,21 @@ object PowerIterationClustering extends Logging {
             s" difference delta = ${diffDelta} and norm = ${norm}")
         }
       }
+      curG.vertices.unpersist()
+      curG.edges.unpersist()
       // update v
       curG = Graph(VertexRDD(v1), g.edges)
+      materialize(curG)
+      v.unpersist()
       prevDelta = delta
     }
-    curG.vertices
+    val eigenVectorRDD = curG.vertices.cache()
+    // materialize the eigen vector RDD and unpersist the graph
+    eigenVectorRDD.count()
+    curG.vertices.unpersist()
```

Comment:
Isn't this unpersisting eigenVectorRDD? they're the same RDD

Reply:
Sorry, updated the code